doc/porting/arch.rst

```diff
@@ -458,3 +458,93 @@ Each architecture also needs its own linker script, even if most sections can
 be derived from the linker scripts of other architectures. Some sections might
 be specific to the new architecture, for example the SCB section on ARM and the
 IDT section on x86.
+
+Hardware Stack Protection
```

Comment:
This is kind of obvious from the description below, but might be a good idea to note that this alone isn't going to stop any bullets.

Reply:
done